ui/admin/app/routes/scopes/scope/roles/role/add-principals.js

```diff
@@ -38,10 +41,39 @@ export default class ScopesScopeRolesRoleAddPrincipalsRoute extends Route {
    * Returns the current role, all users, and all groups
    * @return {{role: RoleModel, users: [UserModel], groups: [GroupModel]}}
    */
-  model() {
+  async model() {
     const role = this.modelFor('scopes.scope.roles.role');
     const scopes = this.store.peekAll('scope').toArray();
     const scopeIDs = this.scope?.map((scope) => scope.id);
+    const authMethods = scopeIDs?.length
+      ? await this.resourceFilterStore.queryBy(
+          'auth-method',
+          {
+            type: 'oidc',
+            scope_id: scopeIDs,
+          },
+          {
+            scope_id: 'global',
+            recursive: true,
+          }
+        )
+      : await this.resourceFilterStore.queryBy(
+          'auth-method',
+          {
+            type: 'oidc',
+          },
+          {
+            scope_id: 'global',
+            recursive: true,
+          }
+        );
+
+    await all(
+      authMethods.map(({ id: auth_method_id }) => {
+        return this.store.query('managed-group', { auth_method_id });
+      })
+    );
+    const managedGroups = this.store.peekAll('managed-group');
```

Comment:
Can this be refactored to load managed groups similar to how it's done within the role model?  That way it's unnecessary to peek all or unload all.

Reply:
yup! working on this refactor